cmd/ingester/app/consumer/consumer.go

```diff
@@ -78,17 +79,11 @@ func (c *Consumer) Start() {
 		c.logger.Info("Starting main loop")
 		for pc := range c.internalConsumer.Partitions() {
 			c.partitionMapLock.Lock()
-			if p, ok := c.partitionIDToState[pc.Partition()]; ok {
```

Comment:
Are you sure this is safe to remove? From the comment is seems like a case where a partition gets re-assigned to the same service instance, so there's a potentially another partition consumer for this partition ID already running (or more likely shutting down).

@vprithvi do you have any more color to add to this condition?

Reply:
We noticed the case that you described a couple of times, but weren't able to reproduce it in tests. 
Since then, many things have changed with both Kafka and Sarama. I think we can remove this now in the interest of simplifying and revisit if see problems.

Reply:
How did it manifest when it happened?  Committed offsets going backwards?

Reply:
IIUC, this happens when new messages arrive on the partition (because the `Messages()` channel isn't closed yet) after the partition consumer was closed (via the `c.closePartition(p.partitionConsumer)` call), causing a new partition consumer to be created.

This PR proposes to solve this by calling `c.internalConsumer.Close()` first within `Close()` which closes the `Messages()` channel, preventing new messages from arriving, and then closes the `Partitions()` channel, preventing new PartitionConsumers from being created.